### ISSUE TITLE
[Feature] WorldThingDamaged: Including Armor

### ISSUE THREAD

**Issue by MajorCooke** [ZDoom/gzdoom#2631]
[Zandronum](https://wiki.zandronum.com/EVENT_scripts#Additional_information) has been including events similar to ZScript's own, and seeing the armor check actually has me wondering if it's possible to expand WorldThingDamaged to include armor - including for 100% absorption. This would allow me to get rid of a few hacks in my own mods that are quite shaky, and I can't help think could be exploited.

If we have something like that for WorldThingDamaged, I could dump those hacks in a heartbeat.

**Comment by Boondorl** [comment]
Would #2485 do what you need?

**Comment by MajorCooke** [comment]
Unfortunately, no. When armor or any damage reduction cancels out damage, WorldThingDamaged is never called. It all falls down to DamageMobj internally to call it.

**Comment by Boondorl** [comment]
Ah, so the issue here is more so that WorldThingDamaged is being skipped in some instances

**Comment by MajorCooke** [comment]
Indeed, and the fact that Zandronum thought of it first to me is a bit of an awakening. Not to mention they have SQL implementation too!

**Comment by LuciferSam8086** [comment]
AFAIK WorldThingDamaged is read-only , so you can't modify such values. 

I did an experiment on my own branch , what if a AbsorbDamageEx / ModifyDamageEx on the item which is always called ?

**Comment by MajorCooke** [comment]
The idea is to modify the engine directly and include a new variable for WorldThingDamaged's struct.
